curvefs/src/client/s3/disk_cache_manager.cpp

```diff
@@ -313,53 +334,69 @@ void DiskCacheManager::SetDiskInitUsedBytes() {
 bool DiskCacheManager::IsDiskCacheFull() {
     int64_t ratio = diskFsUsedRatio_.load();
     uint64_t usedBytes = GetDiskUsedbytes();
-    if (ratio >= fullRatio_ || usedBytes >= maxUsableSpaceBytes_) {
+    if (ratio >= FLAGS_diskFullRatio ||
+        usedBytes >= FLAGS_diskMaxUsableSpaceBytes) {
         VLOG(6) << "disk cache is full"
-                     << ", ratio is: " << ratio << ", fullRatio is: "
-                     << fullRatio_ << ", used bytes is: " << usedBytes;
+                << ", ratio is: " << ratio
+                << ", fullRatio is: " << FLAGS_diskFullRatio
+                << ", used bytes is: " << usedBytes;
         waitIntervalSec_.StopWait();
         return true;
     }
-    if (!IsDiskCacheSafe()) {
+    if (!IsDiskCacheSafe(kRatioLevel)) {
         VLOG(6) << "wake up trim thread.";
         waitIntervalSec_.StopWait();
     }
     return false;
 }
 
-bool DiskCacheManager::IsDiskCacheSafe() {
+bool DiskCacheManager::IsDiskCacheSafe(uint32_t baseRatio) {
     if (IsExceedFileNums()) {
```

Comment:
Would it better to put this check in L337?

Reply:
> Would it better to put this check in L337?

fix